src/main/java/com/rackspace/ceres/app/web/MetadataController.java

```diff
@@ -33,31 +38,44 @@
 
   private final MetadataService metadataService;
 
+  private final Environment environment;
+
   @Autowired
-  public MetadataController(MetadataService metadataService) {
+  public MetadataController(MetadataService metadataService, Environment environment) {
     this.metadataService = metadataService;
+    this.environment = environment;
   }
 
   @GetMapping("/tenants")
   public Mono<List<String>> getTenants() {
-    return metadataService.getTenants();
+    if(isDevProfileActive()) {
+      return metadataService.getTenants();
+    }
+    return Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND));
   }
 
   @GetMapping("/metricNames")
-  public Mono<List<String>> getMetricNames(@RequestParam String tenant) {
-    return metadataService.getMetricNames(tenant);
+  public Mono<List<String>> getMetricNames(@RequestParam(required = false) String tenant,
+      @RequestHeader(value = "X-Tenant-Id", required = false) String tenantId) {
```

Comment:
It would be nice not to hardcode the header name. Can you see if it'll accept the SpEL `"#{appProperties.tenantHeader}"`?